src/Microsoft.Health.Dicom.Api/Features/BackgroundServices/StartBlobDeleteMigrationService.cs

```diff
@@ -0,0 +1,106 @@
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Core.Configs;
+using Microsoft.Health.Dicom.Core.Features.Operations;
+using Microsoft.Health.Dicom.Core.Models.BlobMigration;
+using Microsoft.Health.Dicom.Core.Models.Operations;
+using Microsoft.Health.Operations;
+
+namespace Microsoft.Health.Dicom.Api.Features.BackgroundServices;
+
+/// <summary>
+/// Background service to delete olb blob format after copy operation is successful
+/// </summary>
+public class StartBlobDeleteMigrationService : BackgroundService
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ILogger<StartBlobDeleteMigrationService> _logger;
+    private readonly BlobMigrationConfiguration _blobMigrationFormatConfiguration;
+
+    public StartBlobDeleteMigrationService(
+        IServiceProvider serviceProvider,
+        IOptions<BlobMigrationConfiguration> blobMigrationFormatConfiguration,
+        ILogger<StartBlobDeleteMigrationService> logger)
+    {
+        EnsureArg.IsNotNull(serviceProvider, nameof(serviceProvider));
+        EnsureArg.IsNotNull(blobMigrationFormatConfiguration, nameof(blobMigrationFormatConfiguration));
+        EnsureArg.IsNotNull(logger, nameof(logger));
+
+        _serviceProvider = serviceProvider;
+        _blobMigrationFormatConfiguration = blobMigrationFormatConfiguration.Value;
+        _logger = logger;
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        try
+        {
+            using IServiceScope scope = _serviceProvider.CreateScope();
+
+            // Start the background service only when the flag is turned on and the format type is new.
+            if (_blobMigrationFormatConfiguration.FormatType == BlobMigrationFormatType.New && _blobMigrationFormatConfiguration.StartDelete)
+            {
+                IDicomOperationsClient operationsClient = scope.ServiceProvider.GetRequiredService<IDicomOperationsClient>();
+
+                // Get existing delete operation status
+                OperationCheckpointState<DicomOperation> existingInstance = await operationsClient.GetLastCheckpointAsync(_blobMigrationFormatConfiguration.DeleteOperationId, stoppingToken); ;
+
+                if (existingInstance == null)
+                {
+                    _logger.LogDebug("No existing delete operation.");
+                }
+                else
+                {
+                    _logger.LogDebug("Existing delete operation is in status: '{Status}'", existingInstance.Status);
+                }
+
+                OperationCheckpointState<DicomOperation> copyOperation = await operationsClient.GetLastCheckpointAsync(_blobMigrationFormatConfiguration.CopyOperationId, stoppingToken);
```

Comment:
Shouldn't this line move inside of the first `if` block?

Reply:
Yeah, I think it was changed after the merge conflict